[Core Profiler] Remove plugin limit and improve layout

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes https://github.com/woocommerce/woocommerce/issues/56079

- Revert the changes made in https://github.com/woocommerce/woocommerce/pull/53742 to removes the 8-plugin display limit from the Core Profiler's plugin recommendations
- Improves layout handling using CSS variables and dynamic calculation so the sticky footer behavior to work consistently across different screen sizes
- Limit / truncate descriptions to 1 line, e.g. use CSS to prevent wrapping to additional lines.


(For Bug Fixes) Bug introduced in PR # .

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|  ![Screenshot 2025-03-05 at 15 46 30](https://github.com/user-attachments/assets/d2ea58b0-6733-4763-82c0-0768877598a8)     |    <img  alt="Screenshot 2025-03-06 at 11 19 14" src="https://github.com/user-attachments/assets/fc9b4086-eead-4ed7-9964-9e32612edf4d" />
) |



https://github.com/user-attachments/assets/7413b699-d2d6-495e-93dc-d2ee9d8ed880

https://github.com/user-attachments/assets/4e3a31b7-e4d6-44d4-b999-438ea543bd4a


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site
2. Install and activate Code Snippets plugin
3. Activate the following code snippet:

```php
function filter_free_extensions_response($response) {
    $plugins = array();
    for($i = 1; $i <= 10; $i++) {
        $plugins[] = array(
            "name" => "WooPlugin " . $i,
            "description" => "Plugin " . $i . " Securely accept payments and manage payment activity straight from your store's dashboard",
            "is_visible" => true,
            "is_built_by_wc" => true,
            "key" => "woo-plugin-" . strtolower($i),
            "label" => "Plugin " . $i . " Label",
            "image_url" => "https://woocommerce.com/wp-content/plugins/wccom-plugins/obw-free-extensions/images/core-profiler/logo-woo-rebrand-2.svg",
            "learn_more_link" => "https://woocommerce.com",
            "install_priority" => $i,
            "is_installed" => false,
            "is_activated" => false
        );
    }

    return array(
        array(
            "key" => "obw/core-profiler",
            "title" => "Grow your store",
            "plugins" => $plugins
        )
    );
}

add_filter('rest_pre_echo_response', function($response, $object, $request) {
    if ($request->get_route() === '/wc-admin/onboarding/free-extensions') {
        return filter_free_extensions_response($response);
    }
    return $response;
}, 10, 3);
```

3. Navigate to WooCommerce Core Profiler
4. Proceed to the plugin recommendations page
5. Verify that:
  - All recommended plugins are displayed (no 8-plugin limit)
   - Plugins maintain a 2-column layout
   - Layout adjusts properly for different screen sizes
   - Sticky footer appears/disappears appropriately based on content
   - No visual regressions in plugin card styling
   - Limit / truncate descriptions to 1 line
6. Change the `$plugin_count` variable in the code snippet to various values (just test possible reasonable values) and verify the layout behaves as expected.


<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Remove plugin display limit and improve layout handling in Core Profiler

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan committed:
Remove `pluginsTruncated` references in Core Profiler

This commit removes the `pluginsTruncated` property and related logic from the Core Profiler, simplifying the plugin selection process and ensuring all available plugins are shown. Changes include:
- Removing `pluginsTruncated` from type definitions
- Updating the plugin handling logic to return all available plugins
- Removing references to `pluginsTruncated` in event tracking and component props

chihsuan committed:
Improve sticky footer logic in Core Profiler Plugins page

Refactor the sticky footer implementation for the Plugins page by:
- Simplifying the media query conditions for sticky footer
- Using a dynamic height calculation based on row count
- Adding a new `useMediaQuery` hook to determine when to apply sticky footer
- Updating the SCSS to use a single `.sticky-footer` class instead of multiple row-specific classes

chihsuan committed:
Improve plugin card text styling in Core Profiler

Update the plugin card text SCSS to:
- Ensure labels have max-width of 100%
- Add text truncation for long plugin descriptions
- Use flexbox for better text layout
- Prevent text wrapping and add ellipsis for overflow

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

chihsuan committed:
Merge unused file

chihsuan committed:
Update row height

ilyasfoo commented on plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-card/plugin-card.scss:
```diff
@@ -123,13 +124,24 @@
+			overflow: hidden;
+			text-overflow: ellipsis;
+		}
+
```
Nit
```suggestion
```

ilyasfoo reviewed (commented):
Thanks for working on this, @chihsuan! Looking good so far. Some feedback:
1. The logo and checkbox should be vertically centered ([screenshot](https://github.com/user-attachments/assets/a20a2091-e3d5-4304-8189-0bdd23606ca8))
2. According to the design, the borders should be more rounded
3. Probably not related to the PR, but on specific screen sizes (ex: 940px) the padding   on left side seems to be imbalanced ([screenshot](https://github.com/user-attachments/assets/a9b9939c-8e30-4ec6-9815-cdf9f857fbe3))

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Not applicable |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Testing instructions  | ✅ All good |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result |
| ------------- | ------------- |
| All recommended plugins are displayed (no 8-plugin limit) | ✅ Expected |
| Plugins maintain a 2-column layout | ✅ Expected |
| Layout adjusts properly for different screen sizes | 🟡 Commented |
| Sticky footer appears/disappears appropriately based on content | ✅ Expected |
| No visual regressions in plugin card styling | ✅ Expected |
| Change the $plugin_count variable in the code snippet to various values (just test possible reasonable values) and verify the layout behaves as expected.| ✅ Expected |


#### Extension description

I think this change is fundamentally affecting the extensions' description. It seems down to around 38 characters long ([screenshot](https://github.com/user-attachments/assets/71f888ff-e4ad-45dd-b7ec-a09fa0ae1a24)). The "Learn more" link is also taking up precious space. I think we may need to polish both the presentation and the copywriting on these descriptions (cc @adrianduffell @verofasulo).

moon0326 reviewed (commented):
LGTM and tested well 👍 🚀 

As Ilyas mentioned, I think the description and the Learn More section feel a bit cramped.

@ilyasfoo Ilyas has already provided very detailed comments, so I have nothing to add.

chihsuan committed:
Remove learn more link

chihsuan committed:
Update plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-card/plugin-card.scss

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

chihsuan commented:
> I think this change is fundamentally affecting the extensions' description. It seems down to around 38 characters long ([screenshot](https://github.com/user-attachments/assets/71f888ff-e4ad-45dd-b7ec-a09fa0ae1a24)). The "Learn more" link is also taking up precious space. I think we may need to polish both the presentation and the copywriting on these descriptions

Thanks @ilyasfoo @moon0326, I've removed the  “Learn more” link.  p1741187620609249/1741166258.164319-slack-C01SFMVEYAK

adrianduffell commented:
> Probably not related to the PR, but on specific screen sizes (ex: 940px) the padding on left side seems to be imbalanced ([screenshot](https://github.com/user-attachments/assets/a9b9939c-8e30-4ec6-9815-cdf9f857fbe3))

This looks to be present on older versions too. We can possibly fix it outside this PR 🤔 

<img width="1006" alt="Screen Shot 2025-03-06 at 2 22 09 pm" src="https://github.com/user-attachments/assets/2f4711fb-a262-4f7b-abb6-181e81aaefc1" />

chihsuan committed:
Update plugin card border-radius

chihsuan committed:
Update plugin card to match design

chihsuan committed:
Update row height

chihsuan commented:
> The logo and checkbox should be vertically centered ([screenshot](https://github.com/user-attachments/assets/a20a2091-e3d5-4304-8189-0bdd23606ca8))
According to the design, the borders should be more rounded
Probably not related to the PR, but on specific screen sizes (ex: 940px) the padding on left side seems to be imbalanced ([screenshot](https://github.com/user-attachments/assets/a9b9939c-8e30-4ec6-9815-cdf9f857fbe3))

@ilyasfoo @adrianduffell I've addressed the comments and updated the UI to match the design ( Y5pUYSJPsGEud1vknUZhi8-fi-5399_33846). Please take another look. :pray:

<img width="1147" alt="Screenshot 2025-03-06 at 17 06 15" src="https://github.com/user-attachments/assets/b53a048f-30bc-42e1-b15c-eed181347159" />


https://github.com/user-attachments/assets/7e7cba1d-4b16-46bf-9394-bed965ed2d1f

ilyasfoo reviewed (approved):
Awesome, @chihsuan! LGTM!

<img width="978" alt="image" src="https://github.com/user-attachments/assets/6fe29b22-72f9-4ab2-8cfc-f61d48cdd404" />

#### Scenarios tested

| Scenario | Result |
| ------------- | ------------- |
| The logo and checkbox should be vertically centered | ✅ Works as expected |
| Rounded borders  | ✅ Works as expected |
| Balanced padding | ✅ Works as expected |

adrianduffell committed:
Update header copy

chihsuan committed:
Adjust core profiler plugins page layout and styling

- Increase sticky footer height
- Adjust continue button margin
- Add margin to Jetpack agreement text
- Reduce base height for plugin list

chihsuan committed:
Update text line-height

chihsuan committed:
Improve plugin card responsiveness and styling

- Refactor plugin card layout for better responsiveness
- Add grid layout to plugin list for improved display
- Adjust padding and alignment for different screen sizes
- Enhance description text handling with useMemo
- Add title attribute to description for full text visibility

chihsuan commented:
@adrianduffell  I've made the changes as requested. Thanks for catching the alignment issue.

- Shortened the header copy 6ccc29b377da90d0c7dafbd1defbbfc08c937a0c
- Added more padding to the bottom
86ce05fa4d923fff86aec566ecc41431e833d7de
- Brought the title and descriptions closer together in the recommendation cards to match the design closer 543bcb32608005fa67cfb487c036fc44ecac5590
- Truncate the description to one line for desktop size only. Tablet size therefore wraps text like before and the logo and checkbox should be in the top position / Alignment issue  40a5fc1f3026e27b025f96173ff16a4b52bc323e 


### Desktop

![Screenshot 2025-03-07 at 12 29 03](https://github.com/user-attachments/assets/d4fe0934-0f6d-48fc-9b01-0ecb4faf553f)
![Screenshot 2025-03-07 at 12 28 47](https://github.com/user-attachments/assets/1b614add-19cb-46f0-acc5-4bb0176915bc)

### Tablet
<img src="https://github.com/user-attachments/assets/0be0279d-5aa4-4e9a-a4ef-336486d93986" alt="Screenshot 2025-03-07 at 12 35 06" width="500">

### Mobile
<img src="https://github.com/user-attachments/assets/ba3183fe-418c-4989-a0f5-412c073e07eb" alt="Screenshot 2025-03-07 at 12 35 06" width="500">

### Video

https://github.com/user-attachments/assets/1bf69cac-79e3-49c0-8646-5a9812a8c4b0

adrianduffell commented:
Thanks @chihsuan , overall the changes are looking great! A couple of requests:

* Could the sticky footer have a 24px padding from the edge of the screen. I may be wrong -- it looks like 20px to me. The idea is to double the previous size (12px)
* Could the ipad layout keep the 2-column layout. So it should look something like this:

<img width="1792" alt="tablet logo top" src="https://github.com/user-attachments/assets/c5fd2555-3244-488f-ae54-8b1adc7230e2" />

* You might also be able to set the tablet breakpoint slightly larger. It looks like between 961-999 pixels, the card widths are resized but don't have the copy wrapped on a second line.

chihsuan committed:
Refine core profiler plugin card and list styling

- Adjust max-width and grid layout for plugin cards
- Remove breakpoint for 960px and use media query with $break-xlarge
- Add horizontal padding to plugin list container
- Fine-tune responsive design for plugin cards and list

chihsuan commented:
> Could the sticky footer have a 24px padding from the edge of the screen. I may be wrong -- it looks like 20px to me. The idea is to double the previous size (12px)

@adrianduffell  I checked the Figma final design here (Y5pUYSJPsGEud1vknUZhi8-fi-5512_28249). I believe it needs to be 20px. Let me know if I'm wrong. 🙏 

![Screenshot 2025-03-07 at 14 03 33](https://github.com/user-attachments/assets/d0e5d690-dfa6-416d-b7f8-013ee5540933)

> Could the tablet breakpoint keep the 2-column layout. So it should look something like this:
> You might also be able to set the tablet breakpoint slightly larger. It looks like between 961-999 pixels, the card widths are resized but don't have the copy wrapped on a second line.

Yes, I fixed it in 89b5508d764c66b3c424faad1f408f053b1cf5d4. I had to use custom media queries instead of the existing breakpoint mixins because the current mixins are based on wp-admin breakpoints with limited sizes. I noticed that we are using several different methods to handle breakpoints. I believe we should establish a consistent breakpoint system.

![Screenshot 2025-03-07 at 14 34 36](https://github.com/user-attachments/assets/23764823-e7a0-4828-8947-31a028e4aeb7)

adrianduffell commented:
> @adrianduffell I checked the Figma final design here (Y5pUYSJPsGEud1vknUZhi8-fi-5512_28249). I believe it needs to be 20px. Let me know if I'm wrong. 🙏

@chihsuan Ah apologies, I think the Figma is now out of date for this. I'll see about updating it -- it should be 24.

> I had to use custom media queries instead of the existing breakpoint mixins because the current mixins are based on wp-admin breakpoints with limited sizes. I noticed that we are using several different methods to handle breakpoints. I believe we should establish a consistent breakpoint system.

This is working really great for me now, thank you!

Sounds like it would be worth to establish consistent breakpoints, do you want to create an issue about what you've found?

chihsuan committed:
Change footer margin-bottom to 24px

chihsuan commented:
> Ah apologies, I think the Figma is now out of date for this. I'll see about updating it -- it should be 24.

No worries. I've changed it to 24px in 8875abafb5d9c2dd4afcd7045662c53b2b469222. 🙌 

> Sounds like it would be worth to establish consistent breakpoints, do you want to create an issue about what you've found?

Yeah, I'll look into it a bit more and create an issue for it.

chihsuan committed:
Merge branch 'release/9.8' into update/core-profilter-free-features

chihsuan committed:
Fix test

chihsuan committed:
Fix padding

adrianduffell reviewed (approved):
Testing well, thanks for your quick work here @chihsuan 🚀 

Tested on JN with:

* odd and even amount of recommendations
* existing recommendation copy is truncated on desktop
* 10 recommendations with dummy values 
* desktop / tablet / mobile screens

Screenshot showing current Australian recommendations with truncated copy on desktop.

<img width="1792" alt="Screen Shot 2025-03-08 at 12 44 41 am" src="https://github.com/user-attachments/assets/cee7ac2b-311f-467a-9a4f-c0388ee43c58" />
